### DISCUSSION TITLE
AdGuard install working, but not for my home network

### DISCUSSION THREAD

**Discussion by Gokumon** [AdguardTeam/AdGuardHome#5160] [Q&A]
Hello,

I've recently spun up a VM in Linode (linux, debian 10) and installed adguard. I went through basic configuration and did testing. What I've noticed is any device on my home network will not reach my AdGuard server, even if I've manually set the DNS for the device. I've had friends test, and I immediately get a good amount of traffic and blocked ads, so I think it's safe to say the issue lies somewhere in my home network. Upon further testing, I've managed to get a handful of dns queries to go through my AdGuard server, but a large majority do not. I have several devices using my AdGuard server as their only DNS server. Everything loads without issue, but in the past 12 hours of having devices connected, I've only had 46 DNS queries from my IP. 

Some things to note: I've not done much special configuration with my instance of AdGuard. It's stock. I don't use my ISP's modem, I use my own + a firewall. I've had a friend on the same ISP test, though I haven't had anyone try from behind the same brand of firewall.

Any thoughts or assistance on this would be greatly appreciated, and if there's information I've left out, let me know and I'll get it. Thank you.

**Comment by ppfeufer** [comment]
Make sure that all DNS server addresses point to your AGH instance. Many routers for example provide more than one DNS server, they all have to have the AGH instance IP set. The common misconception of primary and secondary DNS servers is that the first (primary) will be used until it's not reachable and then it switched to the secondary. That's not the case, the DNS servers are randomly used without any "hierarchy". So if you have a secondary DNS server set to, let's say Google DNS, it is entirely possible that your AGH instance won't be used at all.

**Reply by Gokumon** [reply]
> Make sure that all DNS server addresses point to your AGH instance. Many routers for example provide more than one DNS server, they all have to have the AGH instance IP set. The common misconception of primary and secondary DNS servers is that the first (primary) will be used until it's not reachable and then it switched to the secondary. That's not the case, the DNS servers are randomly used without any "hierarchy". So if you have a secondary DNS server set to, let's say Google DNS, it is entirely possible that your AGH instance won't be used at all.

Just to confirm, I've tried using only my instance of AGH as my DNS on my PC. The odd thing is I get a very small and very specific trickle of DNS queries hitting it, but the internet seems to work fine. I've also tried using AG's public DNS servers, but still received ads full force. 

Just to give an idea of the small trickle, over the past 3 days of my DNS being set to my instance of AGH, I've received 79 DNS queries. 76 of those were blocked for being ads, the other 3 were microsoft related and weren't blocked. I would think if I had my DNS set to only my AGH server, traffic would only route through that DNS server and I'd be good to go. The alternative is that it would be having issues, and if that were the case I would be experiencing internet issues, but I'm most definitely not.

My two running theories are either I've screwed something up on my install, and it's not logging my queries but is still passing the traffic (which i'm not sold on because other friends have tested for me and it works great for them), or somehow a second DNS server is being used despite me only specifying one in my static settings. Thoughts?

**Comment by Gokumon** [comment]
Never mind, figured it out. Thanks for the input

**Reply by tom-cor** [reply]
Hey Gokumon, can you share how you figured it out? I'm in the same situation as you pointed out in the beggining.
